core/dbt/clients/yaml_helper.py

```diff
@@ -61,3 +75,7 @@ def load_yaml_text(contents):
             error = str(e)
 
         raise dbt.exceptions.ValidationException(error)
+    except dbt.exceptions.DuplicateYamlKeyException as e:
+        # TODO: We may want to raise an exception instead of a warning in the future.
+        msg = f"{e} {path.searched_path}/{path.relative_path}."
+        dbt.exceptions.warn_or_error(msg, log_fmt=warning_tag("{}"))
```

Comment:
You should use `warn_or_raise` here instead of `warn_or_error`.  That way the exception you throw above bubbles up instead of getting overlaid by the `CompilationException` by `warn_or_error`.  To get the message passed through as `DuplicateYamlKeyException` you can modify the message.  
```
e.msg = f"{e} {path.searched_path}/{path.relative_path}."
dbt.exceptions.warn_or_raise(e, log_fmt=warning_tag("{}"))
```

Reply:
With the new changes I see:

```
$ dbt run                                 
10:59:42  Running with dbt=1.2.0-a1
10:59:42  [WARNING]: Compilation Error
  Compilation Error
    Duplicate 'models' key found in yaml file models/schema.yml.
10:59:43  Found 2 models, 0 tests, 0 snapshots, 0 analyses, 167 macros, 0 operations, 0 seed files, 0 sources, 0 exposures, 0 metrics
10:59:43  
10:59:43  Concurrency: 1 threads (target='dev')
...
```

Wondering if that was what you expected here?